### PR TITLE
Fix Clazy v1.12 errors in 2.4

### PR DESCRIPTION
**Pull request by JoergAtGithub**
I ran Clazy v1.12 on our codebase, and got many errors of the following types:

- https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop-detach.md
- https://github.com/KDE/clazy/blob/master/docs/checks/README-fully-qualified-moc-types.md

This PR fixes them.

### CODE REVIEW

File: src/effects/presets/effectpreset.cpp [hunk 1]
```diff
@@ -53,10 +53,12 @@ EffectPreset::EffectPreset(const EffectSlotPointer pEffectSlot)
     for (int parameterTypeId = 0; parameterTypeId < numTypes; ++parameterTypeId) {
         const EffectParameterType parameterType =
                 static_cast<EffectParameterType>(parameterTypeId);
-        for (const auto& pParameter : pEffectSlot->getLoadedParameters().value(parameterType)) {
+        const auto& loadedParameters = pEffectSlot->getLoadedParameters().value(parameterType);
+        for (const auto& pParameter : loadedParameters) {
```
Comment (Swiftb0y) [inline src/effects/presets/effectpreset.cpp:57]: how does that make a difference?

**Review by Swiftb0y — COMMENTED** [review]
changes LGTM, though this does violate our "no mass reformatting" rule. Do we want to break it here? I'm slightly in favor.

**Comment by JoergAtGithub** [comment]
> changes LGTM, though this does violate our "no mass reformatting" rule.

I don't think that the rule applies here, as this is fixing real code issues and not style issues.

File: src/effects/presets/effectpreset.cpp (hunk 1 above)
Comment (JoergAtGithub) [reply src/effects/presets/effectpreset.cpp:57]: Have a look at https://stackoverflow.com/questions/35811053/using-c11-range-based-for-loop-correctly-in-qt

Comment (Swiftb0y) [reply src/effects/presets/effectpreset.cpp:57]: Ah, I found it in a comment on the solution. you can't call `std::as_const` on an rvalue (which is what I was suggesting). That solves that issue. the `const auto&` is a little weird here tbh (I'd argue just having a `const auto` would make more sense), but it works. I guess.

Comment (JoergAtGithub) [reply src/effects/presets/effectpreset.cpp:57]: Yes, all errors and warnings are gone

**Comment by daschuer** [comment]
That is strange. The issue for my understanding applies only in cases where the object is shared. So it looks like we hit a lot of false positives. Has something changed in this topic?

**Comment by daschuer** [comment]
OK, I found a related bug: https://invent.kde.org/sdk/clazy/-/issues/16 that explains the issue. 
Can you check all cases if making the temporary const is possible? That will also improve constant-correctness and remove some visual clutter.

File: src/broadcast/broadcastmanager.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ BroadcastManager::BroadcastManager(SettingsManager* pSettingsManager,
 
     // Initialize connections list from the current state of BroadcastSettings
     QList<BroadcastProfilePtr> profiles = m_pBroadcastSettings->profiles();
-    for (const BroadcastProfilePtr& profile : profiles) {
+    for (const BroadcastProfilePtr& profile : std::as_const(profiles)) {
```
Comment (daschuer) [inline src/broadcast/broadcastmanager.cpp:46]: For instance here: 

```suggestion
    const QList<BroadcastProfilePtr> profiles = m_pBroadcastSettings->profiles();
    for (const BroadcastProfilePtr& profile : profiles) {
```

**Review by daschuer — CHANGES_REQUESTED** [review]

File: node_modules/.cache/prettier/.prettier-caches/59798cdac4200f82f2c4ebba069ab1c61d895b67.json
Comment (Swiftb0y) [inline node_modules/.cache/prettier/.prettier-caches/59798cdac4200f82f2c4ebba069ab1c61d895b67.json:1]: remove?

File: src/soundio/soundmanagerconfig.cpp [hunk 3]
```diff
@@ -513,9 +514,10 @@ void SoundManagerConfig::loadDefaults(SoundManager* soundManager, unsigned int f
     if (flags & SoundManagerConfig::DEVICES) {
         clearOutputs();
         clearInputs();
-        QList<SoundDevicePointer> outputDevices = soundManager->getDeviceList(m_api, true, false);
+        const QList<SoundDevicePointer> outputDevices =
+                soundManager->getDeviceList(m_api, true, false);
```
Comment (Swiftb0y) [inline src/soundio/soundmanagerconfig.cpp:518]: Have you made sure to double check that these are only applied to instances where you return a `const&`? If they return proper values, making them const may lead to other inefficiencies later (disabling move semantics primarily).

File: node_modules/.cache/prettier/.prettier-caches/59798cdac4200f82f2c4ebba069ab1c61d895b67.json
Comment (JoergAtGithub) [reply node_modules/.cache/prettier/.prettier-caches/59798cdac4200f82f2c4ebba069ab1c61d895b67.json:1]: Upps

File: src/soundio/soundmanagerconfig.cpp (hunk 3 above)
Comment (JoergAtGithub) [reply src/soundio/soundmanagerconfig.cpp:518]: It should be only for the QLists, where Clazy recommended std::as_const and the list is never modified.

File: src/widget/whotcuebutton.cpp [hunk 4]
```diff
@@ -108,7 +108,7 @@ void WHotcueButton::mousePressEvent(QMouseEvent* e) {
 
             CuePointer pHotCue;
             QList<CuePointer> cueList = pTrack->getCuePoints();
```
Comment (daschuer) [inline src/widget/whotcuebutton.cpp:110]: This can also become const

File: src/broadcast/broadcastmanager.cpp [hunk 5]
```diff
@@ -118,10 +118,10 @@ void BroadcastManager::slotControlEnabled(double v) {
         m_pBroadcastEnabled->set(false);
         m_pStatusCO->forceSet(STATUSCO_UNCONNECTED);
         QList<BroadcastProfilePtr> profiles = m_pBroadcastSettings->profiles();
```
Comment (daschuer) [inline src/broadcast/broadcastmanager.cpp:120]: this can also be const

File: src/broadcast/broadcastmanager.cpp [hunk 6]
```diff
@@ -138,7 +138,7 @@ void BroadcastManager::slotProfileRemoved(BroadcastProfilePtr profile) {
 
 void BroadcastManager::slotProfilesChanged() {
     QVector<NetworkOutputStreamWorkerPtr> workers = m_pNetworkStream->outputWorkers();
-    for (const NetworkOutputStreamWorkerPtr& pWorker : workers) {
+    for (const NetworkOutputStreamWorkerPtr& pWorker : std::as_const(workers)) {
```
Comment (daschuer) [inline src/broadcast/broadcastmanager.cpp:141]: This can also be turned into a const

File: src/broadcast/broadcastmanager.cpp [hunk 7]
```diff
@@ -199,7 +199,7 @@ bool BroadcastManager::removeConnection(BroadcastProfilePtr profile) {
 
 ShoutConnectionPtr BroadcastManager::findConnectionForProfile(BroadcastProfilePtr profile) {
     QVector<NetworkOutputStreamWorkerPtr> workers = m_pNetworkStream->outputWorkers();
-    for (const NetworkOutputStreamWorkerPtr& pWorker : workers) {
+    for (const NetworkOutputStreamWorkerPtr& pWorker : std::as_const(workers)) {
```
Comment (daschuer) [inline src/broadcast/broadcastmanager.cpp:202]: This can also be const

File: src/broadcast/broadcastmanager.cpp [hunk 8]
```diff
@@ -220,7 +220,7 @@ void BroadcastManager::slotConnectionStatusChanged(int newState) {
 
     // Collect status info
     QList<BroadcastProfilePtr> profiles = m_pBroadcastSettings->profiles();
-    for (BroadcastProfilePtr profile : profiles) {
+    for (BroadcastProfilePtr profile : std::as_const(profiles)) {
```
Comment (daschuer) [inline src/broadcast/broadcastmanager.cpp:223]: const

**Review by daschuer — CHANGES_REQUESTED** [review]
I have found some more cases where  const works, see comments. The rest LGTM

**Comment by daschuer** [comment]
Thank you. LGTM.

**Pull request merged**